src/arrayUtils.js

```diff
@@ -0,0 +1,68 @@
+if (typeof maybeExport === 'undefined') {
+	function maybeExport () {}
+}
+
+/** @type {function(string, boolean=, number=)} */
+function intArrayFromString(stringy, dontAddNull, length) {
+  var len = length > 0 ? length : lengthBytesUTF8(stringy)+1;
+  var u8array = new Array(len);
+  var numBytesWritten = stringToUTF8Array(stringy, u8array, 0, u8array.length);
+  if (dontAddNull) u8array.length = numBytesWritten;
+  return u8array;
+}
+{{{ maybeExport('intArrayFromString') }}}
+
+function intArrayToString(array) {
+  var ret = [];
+  for (var i = 0; i < array.length; i++) {
+    var chr = array[i];
+    if (chr > 0xFF) {
+#if ASSERTIONS
+      assert(false, 'Character code ' + chr + ' (' + String.fromCharCode(chr) + ')  at offset ' + i + ' not in 0x00-0xFF.');
```

Comment:
Testing and my question below about `is_wasm_only` aside, this is really the only thing left to sort out.

The `#if` block isn't being removed in the JS (so it's not just HTML output that's affected), but after removing it locally everything seems to work as intended. In fact, I'm already using this + the initial version of #5338 to implement wasm in my formerly-asm.js-only libraries in production with no apparent issues.